client/src/controllers/ActionController.test.js

```diff
@@ -199,4 +199,61 @@ describe('ActionController', () => {
       expect(textarea.selectionEnd).toBe(textarea.value.length);
     });
   });
+
+  describe('reset method', () => {
+    beforeEach(async () => {
+      await setup(
+        `<input
+          id="reset-test"
+          value="the default"
+          type="text"
+          data-controller="w-action"
+          data-action="some-event->w-action#reset"
+        />`,
+      );
+    });
+    it('should not update value when event is null', () => {
+      const input = document.getElementById('reset-test');
+
+      input.dispatchEvent(new CustomEvent(''));
+
+      expect(input.value).toBe('the default');
+      expect(input.value).not.toBe('not the default');
+    });
+
+    it('should not change value when event does not have required properties', () => {
+      const input = document.getElementById('reset-test');
+
+      input.dispatchEvent(
+        new CustomEvent('some-event', {
+          detail: { wrongValue: 'the default' },
+        }),
+      );
+
+      expect(input.value).toBe('the default');
+      expect(input.value).not.toBe('not the default');
+    });
+
+    it('should not change value when current value and new value are the same', () => {
```

Comment:
This negative test is useful, we should move this below the 'should change when...'.